perlmod/Fink/PkgVersion.pm

```diff
@@ -5839,6 +5839,10 @@ sub get_perl_dir_arch {
 				# 12.0/13.0 system-perl is 5.30.3, but the only supplied
 				# interpreter is /usr/bin/perl5.30 (not perl5.30.3)
 				$perlcmd = "/usr/bin/arch -%m perl5.30";
+			} elsif ((&version_cmp($perlversion, '=', "5.34.1")) and Fink::Services::get_kernel_vers() == '24') {
```

Comment:
Ugh. this part never got updated with macOS 14 and perl 5.30.3, but at least it still gets caught by the `Fink::Services::get_kernel_vers() >= '21'` elsif.